src/object/obj_tx.c

```diff
@@ -476,8 +523,10 @@ dc_tx_op_end(tse_task_t *task, daos_handle_t th, struct dtx_epoch *req_epoch,
 	if (tx->tx_epoch_task == task) {
 		D_ASSERT(!dtx_epoch_chosen(&tx->tx_epoch));
 		tx->tx_epoch.oe_value = rep_epoch;
+		tx->tx_epoch.oe_flags &= ~DTX_EPOCH_UNCERTAIN;
 		if (tx->tx_epoch.oe_first == 0)
 			tx->tx_epoch.oe_first = tx->tx_epoch.oe_value;
+
```

Comment:
We should not make this change. This epoch has no uncertainty for this operation, but it may have uncertainty for later operations if they access RDGs that do not include the server who chose the epoch.

Reply:
Then when the "DTX_EPOCH_UNCERTAIN" flags can be dropped on the client ?

Reply:
Only when the TX has committed successfully.

Reply:
P.S., even if we don't clear UNCERTAIN upon commit, it would be okay, as we don't export UNCERTAIN flag. But clearing it upon commit, like you did, is fine.